examples/abi_supertraits/src/main.sw

```diff
@@ -0,0 +1,24 @@
+contract;
+
+trait ABIsupertrait {
+    fn foo();
+}
+
+abi MyAbi : ABIsupertrait {
+    fn bar();
+} {
+    fn baz() {
+        Self::foo() // supertrait method usage
+    }
+}
+
+impl ABIsupertrait for Contract {
+    fn foo() {}
```

Comment:
I'd be curious to know what happens if `foo` takes `self` as its first parameter. I suspect we're going to have to disallow that for the time being if `ABIsupertrait` is implemented for a `Contract` since `Contract` is an opaque type with no real semantics. For example, one could write the following:
```rust
impl ABIsupertrait for Contract {
    fn foo(self) {
        std::logging::log(self);
    }
}
```
and I'm not sure if this compiles or how to even define it at this point.

Reply:
Yeah, right now I'm getting the following error:

> "self parameter is not allowed for an ABI method signature"

so `self.foo` looks impossible at the moment.

Reply:
That's good! For now :)